src/items.rs

```diff
@@ -138,17 +138,24 @@ impl Rewrite for ast::Local {
                 );
                 result.push_str(&else_kw);
 
-                let allow_single_line = allow_single_line_let_else_block(&result, block);
+                let max_width =
+                    std::cmp::min(shape.width, context.config.single_line_let_else_max_width());
+                let available_space = max_width.saturating_sub(result.len());
```

Comment:
This arithmetic can be a bit confusing to read. Is there similar functionality somewhere else that can be extracted to a common function? Also, can we test that the width configured corresponds to the actual width of the code exactly (probably not required for now)?

Reply:
The only place I know where we clamp the width between two different widths is when rewriting doc comments, but even then it's slightly different: https://github.com/rust-lang/rustfmt/blob/9386b32f5a72bb276a97461d33aab415314201d8/src/comment.rs#L773-L776

At the point where we perform this check we've already written `let {pat} = {expr} else` into the buffer. And we're trying to see how much room we've got left in order to write the divergent `block` on one line. `shape.width` is going to vary based on how indented we are so we need to determine the available space based on the smaller of `shape.width` and `single_line_let_else_max_width`. If `available_space` hits zero then we know ahead of time that we wont be writing the block on one line.

Reply:
Thanks. Could you add that as a comment?

Reply:
Always happy to add more tests. Are there specific test cases you think I should add? I think adding one with comments in the divergent block would be useful to show that those will always be multi-lined regardless of what you specify for `single_line_let_else_max_width`.

I also want to highlight that I tried to be thorough about the boundary tests when implementing `let-else`. Check out [`tests/target/let_else.rs`](https://github.com/rust-lang/rustfmt/blob/master/tests/target/let_else.rs) when you have a chance and let me know if you think I could expand on whats already there 😁

Reply:
Ah. I didn't see that test file. Looks good then.

Reply:
Great! I'll add that comment a little later. I appreciate you taking the time to look this over. Please let me know if you've got any other thoughts on how to improve the PR

Reply:
@fee1-dead I added the comments that you asked for and I expanded on the test cases. Let me know if you'd like to see any other changes or have any other advice to improve the PR!